### PR TITLE
[BEAM-5071] Replace the bigquery with restful APIs to query dependenc…

### PR DESCRIPTION
**Pull request by yifanzou**
Use maven central REST API to get Java deps release dates.
Use the compatibility checking service to get Python dep release dates.
+R: @chamikaramj 

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 1]
```diff
@@ -105,18 +108,26 @@ def prioritize_dependencies(deps, sdk_type):
...
+        curr_release_date = find_release_time_from_maven_central(group_id, artifact_id, curr_ver)
+        latest_release_date = find_release_time_from_maven_central(group_id, artifact_id, latest_ver)
       else:
         dep_details_url = _PYPI_URL + dep_name
-
+        curr_release_date = find_release_time_from_python_compatibility_checking_service(dep_name, curr_ver)
+        latest_release_date = find_release_time_from_python_compatibility_checking_service(dep_name, curr_ver)
+
+      if not curr_release_date or not latest_release_date:
+        curr_release_date, latest_release_date = query_dependency_release_dates_from_bigquery(bigquery_client,
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: So we are going to keep BQ around as a backup ?

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 2]
```diff
@@ -176,7 +187,77 @@ def compare_dependency_versions(curr_ver, latest_ver):
...
+    artifact_id:
+    version:
+  Return:
+    release date
+  """
+  url = "http://search.maven.org/solrsearch/select?q=g:{0}+AND+a:{1}+AND+v:{2}".format(
+      group_id,
+      artifact_id,
+      version)
+  logging.info('Finding release from maven central')
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Let's include artifact name and version in this log.

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 3]
```diff
@@ -176,7 +187,77 @@ def compare_dependency_versions(curr_ver, latest_ver):
...
+  try:
+    response = request_session_with_retries().get(url)
+    if not response.ok:
+      logging.error("The response status code is not ok: " + str(response.status_code))
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Why both an error and a info log ? Drop error log and change info log to error ?

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 4]
```diff
@@ -176,7 +187,77 @@ def compare_dependency_versions(curr_ver, latest_ver):
...
+      logging.info("Failed finding the release date of {0}:{1} {2}".format(group_id, artifact_id, version))
+      return None
+    response_data = response.json()
+    release_timestamp = response_data['response']['docs'][0]['timestamp']
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Are all these fields guaranteed to be available ?

Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Lets rename to release_timestamp_mills for clarity ?

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 5]
```diff
@@ -176,7 +187,77 @@ def compare_dependency_versions(curr_ver, latest_ver):
...
+  """
+  url = 'http://104.197.8.72/?package={0}=={1}&python-version=2'.format(
+      dep_name,
+      version
+  )
+  logging.info('Finding release time from python compatibility checking service.')
+  try:
+    response = request_session_with_retries().get(url)
+    if not response.ok:
+      logging.error("The response status code is not ok: " + str(response.status_code))
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Ditto. Reduce to one log line ?

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 6]
```diff
@@ -176,7 +187,77 @@ def compare_dependency_versions(curr_ver, latest_ver):
...
+      logging.info("Failed finding the release date of {0} {1}".format(dep_name, version))
+      return None
+    response_data = response.json()
+    release_datetime = response_data['dependency_info'][dep_name]['installed_version_time']
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Ditto (field names guaranteed) ?

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py (hunk 1 above)
Comment (yifanzou) [reply .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Yes. 
1. Need the bigquery table to record release dates of non-MavenCentral packages.
2. The Python compatibility checking services is not mature, 10% of requests will timeout even run out 3 retries.

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py (hunk 2 above)
Comment (yifanzou) [reply .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: done

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py (hunk 3 above)
Comment (yifanzou) [reply .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Done. Merged info log into the error.

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py (hunk 4 above)
Comment (yifanzou) [reply .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Sometimes no. The KeyError will be caught if fields are not exist. Return "None" and use the BigQuery as backup.

Comment (yifanzou) [reply .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Done.

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py (hunk 5 above)
Comment (yifanzou) [reply .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Done.

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py (hunk 6 above)
Comment (yifanzou) [reply .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Same as above.

**Comment by chamikaramj** [comment]
LGTM. Thanks.

Could you fixup into a single commit ?

**Comment by yifanzou** [comment]
Squashed. Thanks for reviewing.

**Pull request merged**